app/code/Magento/ConfigurableProduct/Model/ResourceModel/Attribute/OptionSelectBuilder.php

```diff
@@ -97,6 +101,8 @@ public function getSelect(AbstractAttribute $superAttribute, int $productId, Sco
         )->where(
             'attribute.attribute_id = ?',
             $superAttribute->getAttributeId()
+        )->order(
+            'attribute_opt.sort_order ASC'
```

Comment:
Changes looks good to me.

Please update existing unit test: https://travis-ci.org/magento/magento2/jobs/306413370 and add a separate test method to just check options are returned in correct order.

After that please do an amend commit and force push to the same branch.

Reply:
Yea I already see maybe I add also a integrations test because only PHPUnit test makes not so much sense.

Reply:
True, something involving a real database would be more than appreciated here.